src/gui/qgslayermetadataresultsmodel.h

```diff
@@ -0,0 +1,154 @@
+/***************************************************************************
+  qgslayermetadataresultsmodel.h - QgsLayerMetadataResultsModel
+
+ ---------------------
+ begin                : 1.9.2022
+ copyright            : (C) 2022 by ale
+ email                : [your-email-here]
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSLAYERMETADATARESULTSMODEL_H
+#define QGSLAYERMETADATARESULTSMODEL_H
+
+#include <QAbstractTableModel>
+#include <QObject>
+#include <QThread>
+#include "qgis_gui.h"
+#include "qgsabstractlayermetadataprovider.h"
+
+class QgsFeedback;
+
+
+///@cond private
+
+#ifndef SIP_RUN
+
+/**
+ * The QgsMetadataResultFetcher class fetches query results from a separate thread
+ * WARNING: this class is an implementation detail and it is not part of public API!
+ */
+class QgsMetadataResultsFetcher: public QObject
+{
+    Q_OBJECT
+
+  public:
+
+    //! Constructs a metadata result fetcher.
+    QgsMetadataResultsFetcher( const QgsAbstractLayerMetadataProvider *metadataProvider, const QgsMetadataSearchContext &searchContext, QgsFeedback *feedback );
+
+    //! Start fetching.
+    void fetchMetadata( );
+
+  signals:
+
+    //! Emitted when \a results have been fetched
+    void resultsReady( const QgsLayerMetadataSearchResults &results );
+
+  private:
+
+    const QgsAbstractLayerMetadataProvider *mLayerMetadataProvider = nullptr;
+    QgsMetadataSearchContext mSearchContext;
+    QgsFeedback *mFeedback;
+};
+
+#endif
+
+///@endcond private
+
+/**
+ * The QgsLayerMetadataResultsModel class handles the metadata fetching from
+ * the registered layer metadata providers.
+ *
+ * \ingroup gui
+ * \since QGIS 3.28
+ */
+class GUI_EXPORT QgsLayerMetadataResultsModel : public QAbstractTableModel
```

Comment:
Can you add some tests for this + the proxy model please?

Reply:
done